sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -439,6 +435,9 @@ case class FileSourceScanExec(
       driverMetrics("staticFilesNum") = filesNum
       driverMetrics("staticFilesSize") = filesSize
     }
+    if (relation.partitionSchemaOption.isDefined) {
```

Comment:
This change is not needed, but I keep it here for: 1. avoid duplicated code. 2. make it super safe that we only update the `numPartitions` if table is partitioned.